src/storage/UpdateEdgeProcessor.cpp

```diff
@@ -0,0 +1,382 @@
+                                 &collector);
+    if (ret != kvstore::ResultCode::SUCCEEDED) {
+        return ret;
+    }
+    for (auto& tc : this->tagContexts_) {
+        VLOG(3) << "partId " << partId << ", vId " << edgeKey.src
+                << ", tagId " << tc.tagId_ << ", prop size " << tc.props_.size();
+        ret = this->collectVertexProps(partId,
+                                       edgeKey.src,
+                                       tc.tagId_,
+                                       tc.props_,
+                                       &fcontext,
+                                       &collector);
+        if (ret != kvstore::ResultCode::SUCCEEDED) {
+            return ret;
+        }
+    }
+
+    // the Where clause for filting
+    auto& getters = expCtx_->getters();
+    getters.getSrcTagProp = [&, this] (const std::string& tagName,
+                                       const std::string& prop) -> VariantType {
+        auto it = fcontext.tagFilters_.find(std::make_pair(tagName, prop));
+        if (it == fcontext.tagFilters_.end()) {
+            return std::string("Unknown");
+        }
+        VLOG(1) << "Hit srcProp for tag " << tagName << ", prop "
+                << prop << ", value " << it->second;
+        return it->second;
+    };
+    getters.getEdgeProp = [&, this] (const std::string &prop) -> VariantType {
+        auto res = RowReader::getPropByName(this->filterReader_.get(), prop);
+        CHECK(ok(res));
+        auto val = value(std::move(res));
+        VLOG(1) << "Hit edgeProp for edge prop " << prop << ", value " << val;
+        return val;
+    };
+    if (this->exp_ != nullptr) {
+        auto filterResult = this->exp_->eval();
+        if (!Expression::asBool(filterResult)) {
+            VLOG(1) << "Filter skips the update";
+            return kvstore::ResultCode::ERR_FALSE_FILTER;
+        }
+    }
+
+    // prepare the return(yield clause) record
+    std::vector<VariantType> record(returnColumnsNum_);
+    if (returnColumnsNum_ > 0) {
+        std::string encoded = writer.encode();
+        auto writeSchema = std::make_shared<ResultSchemaProvider>(writer.moveSchema());
+        CHECK_EQ(returnColumnsNum_, writeSchema->getNumFields());
+        auto reader = RowReader::getRowReader(encoded, writeSchema);
+        int64_t index = 0;
+        for (auto& prop : this->edgeContext_.props_) {
+            if (prop.returned_) {
+                auto res = RowReader::getPropByIndex(reader.get(), index);
+                if (!ok(res)) {
+                    VLOG(3) << "Bad value for prop " << prop.prop_.get_name();
+                    break;
+                }
+                record[prop.retIndex_] = value(std::move(res));
+                index++;
+            }
+        }
+        for (auto& tc : tagContexts_) {
+            for (auto& prop : tc.props_) {
+                if (prop.returned_) {
+                    auto res = RowReader::getPropByIndex(reader.get(), index);
```

Comment:
maybe here can not reuse index if break from line 157.

Reply:
return kvstore::ResultCode::ERR_FALSE_UPDATER;